File: common/src/main/java/com/cainsgl/common/exception/BSystemException.java
package com.cainsgl.common.exception;

import lombok.Getter;

@Getter
public class BSystemException extends RuntimeException
{
    private static final Integer DEFAULT_ERROR_CODE = 501;
    private final Integer errorCode;
    public BSystemException(String message, Integer errorCode)
    {
        super(message);
        this.errorCode = errorCode;
    }

    public BSystemException(String message)
    {
        this(message, DEFAULT_ERROR_CODE);
    }
}

File: common/src/main/java/com/cainsgl/common/exception/BSystemException.kt
package com.cainsgl.common.exception

class BSystemException : RuntimeException {
    private val errorCode: Int

    constructor(message: String, errorCode: Int) : super(message) {
        this.errorCode = errorCode
    }

    constructor(message: String) : this(message, DEFAULT_ERROR_CODE)
    constructor(ex:Exception) : super(ex) {
        this.errorCode = DEFAULT_ERROR_CODE
    }
    companion object {
        private const val DEFAULT_ERROR_CODE = 501
    }
}
